msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+                    userName,
+                    uid + "." + utid,
+                    new ArrayList<>(Arrays.asList(
+                        new UserAccountData(foreignCloud, randomUuidString(), randomUuidString()),
+                        new UserAccountData(foreignCloud, randomUuidString(), randomUuidString())
+                    ))
+                )},
+            new Object[]{
+                "User signed into Home tenant ",
+                new TestCaseData(
+                    userName,
+                    uid + "." + utid,
+                    new ArrayList<>(Arrays.asList(
+                        new UserAccountData(homeCloud, uid, utid)
+                    ))
+                )},
+            new Object[]{
+                "User signed into Home tenant and 1 guest tenant in home cloud ",
+                new TestCaseData(
+                    userName,
+                    uid + "." + utid,
+                    new ArrayList<>(Arrays.asList(
+                        new UserAccountData(homeCloud, uid, utid),
+                        new UserAccountData(homeCloud, randomUuidString(), randomUuidString())
+                    ))
+                )},
+            new Object[]{
+                "User signed into multiple tenants in home cloud and foreign cloud",
+                new TestCaseData(
+                    userName,
+                    uid + "." + utid,
+                    new ArrayList<>(Arrays.asList(
+                        new UserAccountData(homeCloud, uid, utid),
+                        new UserAccountData(homeCloud, randomUuidString(), randomUuidString()),
+                        new UserAccountData(foreignCloud, randomUuidString(), randomUuidString()),
+                        new UserAccountData(foreignCloud, randomUuidString(), randomUuidString())
+                    ))
+                )}
+        );
+    }
+
+    private static String randomUuidString() {
+        return UUID.randomUUID().toString();
+    }
+
+    @Before
+    public void setup() {
+        super.setup();
+        mMultipleAccountPCA = (IMultipleAccountPublicClientApplication) mApplication;
+        performInteractiveAcquireTokenWithMockResponsesToSetupCache();
+    }
+
+    @Test
+    public void testGetAccountsReturnsSingleAccountObjectForAccountRecordsFromMultipleCloudsWithSameHomeAccountId() {
+        // arrange
+        final ResultFuture<AsyncResult<List<IAccount>>> future = new ResultFuture<>();
+
+        // act
+        mMultipleAccountPCA.getAccounts(new IPublicClientApplication.LoadAccountsCallback() {
+            @Override
+            public void onTaskCompleted(List<IAccount> result) {
+                future.setResult(new AsyncResult<List<IAccount>>(result, null));
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                future.setResult(new AsyncResult<List<IAccount>>(null, exception));
+            }
+        });
+
+        RoboTestUtils.flushScheduler();
```

Comment:
Seems we are using both future and flush scheduler in the same test - seems weird. I believe only one of these paradigms should be needed.

Reply:
I removed the future in latest iteration